src/libmongoc/tests/test-mongoc-primary-stepdown.c

```diff
@@ -65,12 +66,10 @@ _connection_count (mongoc_database_t *db)
    bool res;
    int conns;
 
-   ASSERT (db);
-
    BSON_APPEND_INT32 (&cmd, "serverStatus", 1);
 
-   read_prefs = mongoc_read_prefs_new (MONGOC_READ_PRIMARY);
-   res = mongoc_database_command_simple (db, &cmd, read_prefs, &reply, &error);
+   res = mongoc_client_command_simple_with_server_id (
+      client, "admin", &cmd, NULL, server_id, &reply, &error);
```

Comment:
If I understand your previous comment correctly, the fix here was to use the same `server_id` instead of re-selecting with a primary read preference, correct?